### PR TITLE
Feature/prime 1566 view email/toa

### PR DESCRIPTION
**Pull request by hogmann**
Going to touch up some of the front end test files tomorrow as a lot of components added but tests haven't been updated.

### CODE REVIEW

File: prime-angular-frontend/src/app/modules/adjudication/shared/components/notification-email-view/notification-email-view.component.html [hunk 1]
```diff
@@ -0,0 +1,20 @@
+<app-page-subheader *ngIf="emailTemplate"
+                    class="mb-4"
+                    [divider]="false">
+  <ng-container appPageSubheaderTitle>
+    {{ EmailTemplateTypeEnum[emailTemplate.emailType] | replace: '_':' ' | capitalize: true }}
+  </ng-container>
+  <ng-container appPageSubheaderSummary>
+    This is what the "{{ EmailTemplateTypeEnum[emailTemplate.emailType] | replace: '_':' ' | capitalize: true }}" email
```
Comment (mtpultz) [inline prime-angular-frontend/src/app/modules/adjudication/shared/components/notification-email-view/notification-email-view.component.html:8]: Feels like maybe the subheader should be `Preview of the {{ ... }} email notification` and then you probably don't need the second subheader.

File: prime-angular-frontend/src/app/modules/adjudication/pages/enrollee-email-notification-list-page/enrollee-email-notification-list-page.component.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  public onBack() {
+    this.routeUtils.routeRelativeTo(['../', AdjudicationRoutes.NOTIFICATION_EMAILS]);
+  }
+
+  ngOnInit(): void {
+    this.getEmailTemplates();
+  }
+
+  private getEmailTemplates() {
+    this.busy = this.emailTemplateResource.getEmailTemplates().subscribe(templates => this.templates = templates);
```
Comment (mtpultz) [inline prime-angular-frontend/src/app/modules/adjudication/pages/enrollee-email-notification-list-page/enrollee-email-notification-list-page.component.ts:37]: We should be putting the subscribe on the next line to keep things readable.  I tend to start thinking about my line lengths once I exceed 60-70 characters and wrap unless the are something like a string that is less than 120 characters.

File: prime-angular-frontend/src/app/modules/adjudication/pages/enrollee-email-notification-list-page/enrollee-email-notification-list-page.component.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    private router: Router
+  ) {
+    this.routeUtils = new RouteUtils(route, router, AdjudicationRoutes.routePath(AdjudicationRoutes.NOTIFICATION_EMAILS));
+  }
+
+  public onBack() {
+    this.routeUtils.routeRelativeTo(['../', AdjudicationRoutes.NOTIFICATION_EMAILS]);
+  }
+
+  ngOnInit(): void {
```
Comment (mtpultz) [inline prime-angular-frontend/src/app/modules/adjudication/pages/enrollee-email-notification-list-page/enrollee-email-notification-list-page.component.ts:32]: Just a reminder the more public accessors we don't include the more updates we'll have to do after our next set of Angular updates.

File: prime-angular-frontend/src/app/modules/adjudication/pages/enrollee-email-notification-list-page/enrollee-email-notification-list-page.component.ts [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+  constructor(
+    private emailTemplateResource: EmailTemplateResourceService,
+    private route: ActivatedRoute,
+    private router: Router
+  ) {
+    this.routeUtils = new RouteUtils(route, router, AdjudicationRoutes.routePath(AdjudicationRoutes.NOTIFICATION_EMAILS));
+  }
+
+  public onBack() {
```
Comment (mtpultz) [inline prime-angular-frontend/src/app/modules/adjudication/pages/enrollee-email-notification-list-page/enrollee-email-notification-list-page.component.ts:28]: Just a reminder the methods we have with no return type even `: void` the more updates we'll have to do after our next set of Angular updates

File: prime-angular-frontend/src/app/modules/adjudication/adjudication-routing.module.ts [hunk 5]
```diff
@@ -57,6 +63,31 @@ const routes: Routes = [
             component: EnrolleeBannerPageComponent,
             data: { title: 'Enrollee Banner' }
           },
+          {
```
Comment (mtpultz) [inline prime-angular-frontend/src/app/modules/adjudication/adjudication-routing.module.ts:66]: Feels like we can do something to reduce the number of components.  They should have different routes, but based on the route the page component could have something slightly different displayed like `/enrolee` filters on enrollee emails.  It looks like some of these pages are almost exactly the same.

File: prime-angular-frontend/src/app/modules/adjudication/shared/components/maintenance-container/maintenance-container.component.html [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+        <mat-card-content>
+          View the notification emails.
+        </mat-card-content>
+        <mat-card-actions>
+          <button mat-stroked-button
+                  type="button"
+                  color="primary"
+                  (click)="onRoute(AdjudicationRoutes.NOTIFICATION_EMAILS)">
+            View
+          </button>
```
Comment (johnnyl6) [inline prime-angular-frontend/src/app/modules/adjudication/shared/components/maintenance-container/maintenance-container.component.html:20]: Was trying as a non-superadmin role, could see the View button on frontend but got backend 403 forbidden error after clicking view button. Feels like need to only enable the button for superadmin.

File: prime-angular-frontend/src/app/modules/adjudication/pages/enrollee-email-notification-list-page/enrollee-email-notification-list-page.component.ts (hunk 2 above)
Comment (hogmann) [reply prime-angular-frontend/src/app/modules/adjudication/pages/enrollee-email-notification-list-page/enrollee-email-notification-list-page.component.ts:37]: no
read
it 
and 
weap

File: prime-angular-frontend/src/app/modules/adjudication/shared/components/maintenance-container/maintenance-container.component.html (hunk 6 above)
Comment (hogmann) [reply prime-angular-frontend/src/app/modules/adjudication/shared/components/maintenance-container/maintenance-container.component.html:20]: Have a note i forgot to enable the back end for all admin types have it restricted. so changes coming down

**Comment by neophyte57** [comment]
Should clicking on the "Back" navigation return the user to the list of Notification Emails?

image

File: prime-dotnet-webapi/prime.csproj [hunk 7]
```diff
@@ -30,13 +30,15 @@
     <PackageReference Include="SoapCore" Version="1.1.0.2" />
     <PackageReference Include="Swashbuckle.AspNetCore" Version="5.0.0" />
     <PackageReference Include="Swashbuckle.AspNetCore.Newtonsoft" Version="5.0.0" />
-    <PackageReference Include="CompareNETObjects" Version="4.73.0" />
+    <PackageReference Include="CompareNETObjects" Version="4.65.0" />
```
Comment (james-hollinger) [inline prime-dotnet-webapi/prime.csproj:33]: This seems unintentional

File: prime-dotnet-webapi/prime.csproj [hunk 8]
```diff
@@ -30,13 +30,15 @@
...
     <PackageReference Include="Microsoft.Extensions.Http" Version="3.1.4" />
     <PackageReference Include="Wkhtmltopdf.NetCore" Version="3.0.0" />
     <PackageReference Include="IdentityModel" Version="4.3.0" />
     <PackageReference Include="LinqKit.Microsoft.EntityFrameworkCore" Version="3.0.0" />
     <PackageReference Include="DelegateDecompiler.EntityFrameworkCore" Version="0.28.0" />
     <PackageReference Include="Flurl" Version="3.0.1" />
+    <PackageReference Include="RazorEngine.NetCore" Version="3.1.0" />
+    <PackageReference Include="Microsoft.CodeAnalysis.CSharp.Workspaces" Version="3.6.0" />
```
Comment (james-hollinger) [inline prime-dotnet-webapi/prime.csproj:41]: What is this package reference for?

File: prime-dotnet-webapi/ApiDbContext.cs [hunk 9]
```diff
@@ -176,6 +177,10 @@ protected override void OnModelCreating(ModelBuilder modelBuilder)
             modelBuilder.Entity<Admin>()
                 .HasIndex("UserId")
                 .IsUnique();
+
+            modelBuilder.Entity<EmailTemplate>()
+                .HasIndex("EmailType")
+                .IsUnique();
```
Comment (james-hollinger) [inline prime-dotnet-webapi/ApiDbContext.cs:183]: Would prefer to see DB configuration in a Configuration file,

File: prime-dotnet-webapi/Controllers/AgreementsController.cs [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+        }
+
+        /// <summary>
+        /// Get a list of the latest enrollee Agreement Versions
+        /// </summary>
+        [HttpGet("enrollee/latest", Name = nameof(GetLatestEnrolleeAgreementVersions))]
+        [Authorize(Roles = Roles.ViewEnrollee)]
+        [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+        [ProducesResponseType(StatusCodes.Status403Forbidden)]
+        [ProducesResponseType(typeof(ApiResultResponse<IEnumerable<AgreementVersionViewModel>>), StatusCodes.Status200OK)]
```
Comment (james-hollinger) [inline prime-dotnet-webapi/Controllers/AgreementsController.cs:35]: should be `typeof(ApiResultResponse<IEnumerable<AgreementVersionListViewModel>>)`

File: prime-dotnet-webapi/Controllers/AgreementsController.cs [hunk 11]
```diff
@@ -0,0 +1,58 @@
...
+            var agreements = await _agreementService.GetLatestEnrolleeAgreementVersionsAsync();
+            return Ok(ApiResponse.Result(agreements));
+        }
+
+        /// /api/agreements/enrollee/2
+        /// <summary>
+        /// Get an enrollee Agreement Version by id
+        /// </summary>
+        /// <param name="agreementId"></param>
+        [HttpGet("enrollee/{agreementId}", Name = nameof(GetEnrolleeAgreementVersionById))]
```
Comment (james-hollinger) [inline prime-dotnet-webapi/Controllers/AgreementsController.cs:47]: This gets the `AgreementVersion` regardless if it's an Enrollee Agreement or not. I feel this URL should just be /api/agreements/{agreementId}

File: prime-dotnet-webapi/prime.csproj (hunk 8 above)
Comment (hogmann) [reply prime-dotnet-webapi/prime.csproj:41]: The razor engine need a version defined or it get dependency loading errors when you restore

File: prime-dotnet-webapi/Controllers/EmailsController.cs [hunk 12]
```diff
@@ -52,5 +56,38 @@ public async Task<ActionResult> SendEnrolleeRenewalEmails()
...
+
+        // GET: api/emails/management/templates
+        /// <summary>
+        /// Get email templates
+        /// </summary>
+        [HttpGet("management/templates", Name = nameof(GetEmailTemplates))]
+        [Authorize(Roles = Roles.ViewEnrollee + "," + Roles.ViewSite)]
+        [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+        [ProducesResponseType(StatusCodes.Status403Forbidden)]
+        [ProducesResponseType(typeof(ApiResultResponse<EmailTemplateListViewModel>), StatusCodes.Status200OK)]
```
Comment (james-hollinger) [inline prime-dotnet-webapi/Controllers/EmailsController.cs:70]: should be `typeof(ApiResultResponse<IEnumerable<EmailTemplateListViewModel>>)`C#

**Pull request closed without merging**